xip/xipether.c

```diff
@@ -0,0 +1,449 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <assert.h>
+#include <sys/types.h>
+#include <asm/byteorder.h>
+#include <asm-generic/errno-base.h>
+#include <net/xia_fib.h>
+#include <xia_socket.h>
+#include <sys/socket.h>
+#include <linux/netdevice.h>
+
+#include "xip_common.h"
+#include "utils.h"
+#include "ll_map.h"
+#include "libnetlink.h"
+#include "xiart.h"
+
+#define IFINDEX_STR_SIZE 8
+#define XIA_LLADDR_LEN 12
+#define APPEND_ZEROES 20
+
+static int usage(void)
+{
+	fprintf(stderr,
+"Usage: xip ether { addinterface | delinterface } dev DEV\n"
+"       xip ether { addneigh | delneigh } lladdr LLADDR dev DEV\n"
+"       xip ether show { interfaces | neighs }\n"
+"where  LLADDR := HEXDIGIT{1,2} (':' HEXDIGIT{1,2})*\n"
+"       DEV := STRING NUMBER\n");
+	return -1;
+}
+
+static int form_ether_xid(unsigned oif, unsigned char *lladdr, unsigned tlen,
+			  char *id)
+{
+	char bytes[4];
+	unsigned append = APPEND_ZEROES;
+	int rc;
+
+	bytes[0] = (oif >> 24) & 0xFF;
+	bytes[1] = (oif >> 16) & 0xFF;
+	bytes[2] = (oif >> 8) & 0xFF;
+	bytes[3] = oif & 0xFF;
+	rc = snprintf(id, (tlen + 1),
```

Comment:
Since we changed the third parameter in calls to `form_ether_xid()`, we can make the second parameter here just `tlen`.

Reply:
What should be the third paramter changed to in the function call, @cjdoucette ? The function calling `form_ether_xid()` expects a string containing the ether-xid if rc>0.

Reply:
What should be the third paramter changed to in the function definition, @cjdoucette ? The function calling form_ether_xid() expects a string containing the ether-xid if rc>0.

Reply:
I meant that in the rest of the file below, I suggest that we change the value we are passing in for the third parameter, `tlen`. Once we do that, this expression should be just `tlen`, not `(tlen + 1)`.